subsys/net/lib/sockets/sockets_select.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/time.h>
+
+#include <net/socket.h>
+
+static int *fdset_find_fd(int fd, zsock_fd_set *set)
+{
+	int i, left = set->active;
+
+	for (i = 0; left > 0 && i < ARRAY_SIZE(set->fds); i++) {
+		if (set->fds[i] == fd) {
+			return &set->fds[i];
+		}
+
+		if (set->fds[i] != -1) {
+			left--;
+		}
+	}
+
+	return NULL;
+}
+
+void ZSOCK_FD_ZERO(zsock_fd_set *set)
+{
+	int i;
+
+	set->active = 0;
+
+	for (i = 0; i < ARRAY_SIZE(set->fds); i++) {
+		set->fds[i] = -1;
+	}
+}
+
+int ZSOCK_FD_ISSET(int fd, zsock_fd_set *set)
+{
+	int *p = fdset_find_fd(fd, set);
+
+	return p != NULL;
+}
+
+void ZSOCK_FD_CLR(int fd, zsock_fd_set *set)
+{
+	int *p = fdset_find_fd(fd, set);
+
+	if (p != NULL) {
+		*p = -1;
+		set->active--;
+	}
+}
+
+void ZSOCK_FD_SET(int fd, zsock_fd_set *set)
+{
+	int i;
+
+	if (set->active == ARRAY_SIZE(set->fds)) {
+		/* No room, no way to indicate that. */
+		return;
+	}
+
+	for (i = 0; i < ARRAY_SIZE(set->fds); i++) {
+		if (set->fds[i] == -1) {
+			set->fds[i] = fd;
+			set->active++;
+			return;
+		}
+	}
+}
+
+static int fd_set_to_pollfd(int poll_flags, zsock_fd_set *set,
+			    struct pollfd *pfds, int *pfds_idx)
```

Comment:
Shouldn't `*pfds_idx` be updated somewhere in here? I see that only local copy is incremented. This way the `num_pfds` from `zsock_select` will remain 0, hence making the `zsock_poll` call meaningless.

Reply:
Ack, that's bug, will fix.

Reply:
> Shouldn't *pfds_idx be updated somewhere in here?

Fixed.